ai_ref_kits/automated_self_checkout/directrun.py

```diff
@@ -0,0 +1,487 @@
+                                removed_object_str = f"{v} {k} removed from zone"
+                                removed_action_str = intersecting_bboxes(
+                                    bboxes, person_bbox, removed_object_str)
+                                if removed_action_str is not None:
+                                    # Add the purchased items to a "receipt" of sorts
+                                    if removed_object_str not in purchased_items:
+                                        # print(f"{v} {k}", a_items)
+                                        # if f"{v} {k}" in a_items:
+                                        purchased_items.add(f"{v} {k}")
+                                        p_items.append(f" - {v} {k}")
+                                # Draw the result on the screen
+                                plog(logtable, removed_action_str, k, "remove")
+
+                    if len(added_objects) >= 1:
+                        # An object has been added
+                        for k, v in added_objects.items():
+                            # For each of the objects, check the IOU between a designated object
+                            # and a person.
+                            if 'person' not in k:
+                                added_object_str = f"{v} {k} added to zone"
+                                added_action_str = intersecting_bboxes(
+                                    bboxes, person_bbox, added_object_str)
+                                if added_action_str is not None:
+                                    if added_object_str not in added_items:
+                                        added_items.add(added_object_str)
+                                        a_items.append(added_object_str)
+
+                                # Draw the result on the screen
+                                plog(logtable, added_action_str, k, "add")
+
+                output_video.write(frame)
+                yield gr.skip(), pd.DataFrame(logtable, columns=['time', 'action', 'class', 'message']), toList(purchased_items), getPandasDF(item_list)
+
+            batch = []  # Restart the batch
+            output_video.release()
+
+            # return gr.skip() (instead of a value) will keep the component wihout any change
+            yield output_video_name, pd.DataFrame(logtable, columns=['time', 'action', 'class', 'message']), toList(purchased_items), getPandasDF(item_list)
+
+            output_video_name = f"output_{uuid.uuid4()}.mp4"
+            output_video = cv2.VideoWriter(
+                output_video_name, video_codec, desired_fps, (width, height))  # type: ignore
+
+        iterating, frame = cap.read()
+        n_frames += 1
+
+# Open a the browser using the informed URL
+
+
+def open_browser(url):
+    webbrowser.open_new_tab(url)
+
+
+###########################################################
+print("Starting the Automated Self-Checkout Demo...")
```

Comment:
I'm not a big fan of the code without function. Please wrap in in the main function, provide input arguments (if necessary) and call main from __main__ as it's done in e.g. https://github.com/openvinotoolkit/openvino_build_deploy/blob/master/ai_ref_kits/intelligent_queue_management/app.py